src/dust/coagulation/coagulation.cpp

```diff
@@ -0,0 +1,308 @@
+//========================================================================================
+// (C) (or copyright) 2023-2024. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+// Artemis includes
+#include "dust/coagulation/coagulation.hpp"
+#include "artemis.hpp"
+#include "dust/dust.hpp"
+#include "geometry/geometry.hpp"
+
+using namespace parthenon::package::prelude;
+
+namespace Dust {
+namespace Coagulation {
+//----------------------------------------------------------------------------------------
+//! \fn  StateDescriptor Coagulalation::Initialize
+//! \brief Adds intialization function for coagulation package
+std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
+  auto coag = std::make_shared<StateDescriptor>("coagulation");
+  Params &params = coag->AllParams();
+
+  CoagParams cpars;
+  const int nm = pin->GetOrAddInteger("dust", "nspecies", 1);
+  cpars.nm = nm;
+  cpars.vfrag = pin->GetOrAddReal("dust", "vfrag", 1.e3); // cm/s
```

Comment:
Seeing lots of "unit"-ed quantities here.  Should this MR wait on the units MR @brryan?

Reply:
Now that #24 is merged, "unit"-ed quantities like here should be updated to (1) have a specified unit system for input parameters, e.g. `cm./s` as here (2) converted to code units at the first opportunity, so for example this line should be
```c++
cpars.vfrag = pin->GetOrAddReal("dust", "vfrag", 1.e3) * units.GetSpeedPhysicalToCode; // cm/s
```
where for `Initialize` functions like this `units` and `constants` must be added to the function argument (e.g. `if (do_gas) packages.Add(Gas::Initialize(pin.get(), units, constants));`), whereas for tasks you can pull the `units` object out of the `artemis` package's params.